### ISSUE TITLE
Defaults for propagated and flavor-related attributes on CIM objects and in tupleparser

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#1039]
The current pywbem code is inconsistent in the default values for certain boolean attributes of certain CIM objects.

Summary of the current behavior:

* For CIM objects created by pywbem users: If an attribute is not specified to the ctor, it defaults to the following values:

  - `propagated` attribute of `CIMProperty` defaults to `None`.
  - `propagated` attribute of `CIMMethod` defaults to `False` (not consistent with proposal below).
  - `propagated` attribute of `CIMQualifier` defaults to `None`.
  - flavor-related attributes of `CIMQualifier` default to `None`.
  - flavor-related attributes of `CIMQualifierDeclaration` default to `None`.

* For CIM objects returned from CIM operations: If an attribute is not present in the received CIM-XML, the corresponding attribute of the CIM object defaults to the following values (by means of code setting it that way in the tupleparser):

  - `propagated` in `CIMProperty` defaults to `False` (for all 3 kinds of CIM-XML elements representing properties).
  - `propagated` in `CIMMethod` defaults to `False`.
  - `propagated` in `CIMQualifier` defaults to `None` (not consistent with proposal below).
  - flavor-related attributes in `CIMQualifier` default to `None` (not consistent with proposal below).
  - flavor-related attributes in `CIMQualifierDeclaration` default to `None` (not consistent with proposal below).

Some facts:

* The CIM objects define that `None` on these attributes means that the information is not available.

* DSP0201 defines defaults for these attributes when omitted in the CIM-XML, as follows:

  * PROPAGATED: false
  * OVERRIDABLE: true
  * TOSUBCLASS: true
  * TOINSTANCE: false
  * TRANSLATABLE: false

Proposal:

* CIM objects created by pywbem users should default these attributes to `None` because that reflects the meaning of "information not available".
* CIM objects created from CIM-XML responses should default these attributes to their DSP0201-defined values.
* As a consequence, the setting-up of expected CIM objects for comparing CIM operation results becomes a bit more work intensive because these attributes now need to be specified.

**Discussion point:** Decide how to handle the defaults in these cases.

**Comment by andy-maier** [comment]
We discussed compatibility of the proposed changes and came to the conclusion that all changes can be made: 
* For CIM objects returned by operations, setting these attributes to the DTD-defined defaults if omitted by the server is not distinguishable by the user from the server having specified them, and they are on the right defaults.
* For CIm objects created by pywbem users, the only change is the default for the propagated attribute of methods, and that ought to be ignored by the server anyway in all operations that pass in methods.

Bottom line: We agreed to go ahead with the proposal.

**Comment by andy-maier** [comment]
Addressed in PR #1076.
